pkg/build/controller/strategy/custom.go

```diff
@@ -40,13 +41,19 @@ func (bs *CustomBuildStrategy) CreateBuildPod(build *buildapi.Build) (*kapi.Pod,
 	}
 
 	if strategy.ExposeDockerSocket {
-		glog.V(2).Infof("ExposeDockerSocket is enabled for %s build", build.PodName)
+		glog.V(2).Infof("ExposeDockerSocket is enabled for %s build", build.Name)
 		containerEnv = append(containerEnv, kapi.EnvVar{"DOCKER_SOCKET", dockerSocketPath})
 	}
 
+	// Need to copy the build object here so that it doesn't share pointers with
+	// the original build object, needed for modyfing labels
```

Comment:
Why don't you just copy the labels of the old build into a new map and instead keep a copy of the whole build?